website/addons/github/static/githubNodeConfig.js

```diff
@@ -0,0 +1,456 @@
+        self.importAuth.call(self);
+    };
+    window.open(self.urls().auth);
+};
+
+ViewModel.prototype.connectExistingAccount = function(account_id) {
+    var self = this;
+    return $osf.putJSON(
+        self.urls().importAuth, {
+            external_account_id: account_id
+        }
+    ).done(function(response) {
+            debugger;
+        self.changeMessage('Successfully imported Github credentials.', 'text-success');
+        self.updateFromData(response);
+
+    }).fail(function(xhr, status, error) {
+        var message = 'Could not import Github credentials at ' +
+            'this time. Please refresh the page. If the problem persists, email ' +
+            '<a href="mailto:support@osf.io">support@osf.io</a>.';
+        self.changeMessage(message, 'text-warning');
+        Raven.captureMessage('Could not import Github credentials', {
+            url: self.urls().importAuth,
+            textStatus: status,
+            error: error
+        });
+    });
+};
+
+ViewModel.prototype._deauthorizeNodeConfirm = function() {
+    var self = this;
+    return $.ajax({
+        type: 'DELETE',
+        url: self.urls().deauthorize,
+        contentType: 'application/json',
+        dataType: 'json'
+    }).done(function(response) {
+        self.updateFromData(response);
+        self.changeMessage('Successfully deauthorized GitHub credentials', 'text-success');
+    }).fail(function(xhr, status, error) {
+        var message = 'Could not deauthorize Github at ' +
+            'this time. Please refresh the page. If the problem persists, email ' +
+            '<a href="mailto:support@osf.io">support@osf.io</a>.';
+        self.changeMessage(message, 'text-warning');
+        Raven.captureMessage('Could not remove Github authorization.', {
+            url: self.urls().deauthorize,
+            textStatus: status,
+            error: error
+        });
+    });
+};
+
+ViewModel.prototype.deauthorizeNode = function() {
+    var self = this;
+    bootbox.confirm({
+        title: 'Deauthorize Github?',
+        message: 'Are you sure you want to remove this Github authorization?',
+        callback: function(confirm) {
+            if (confirm) {
+                self._deauthorizeNodeConfirm();
+            }
+        }
+    });
+};
+
+ViewModel.prototype.updateRepoList = function() {
+    var self = this;
+    return self.fetchRepoList()
+        .done(function(repos) {
+            self.repoList(repos);
+            self.selectedRepo(self.currentRepo());
+        });
+};
+
+ViewModel.prototype.createRepo = function(repoName) {
+    var self = this;
+    self.creating(true);
+    repoName = repoName.toLowerCase();
+    return $osf.postJSON(
+        self.urls().create_repo, {
+            repo_name: repoName
+        }
+    ).done(function(response) {
+        self.creating(false);
+        self.repoList(response.repos);
+        self.loadedRepoList(true);
+        self.selectedRepo((self.ownerName() + " / " + repoName));
```

Comment:
get_repo_drop_down is doing the formatting currently because its possible for the "ownerName" part of the formatting to be a different user (ie a GitHub repo you have access to but is not in git URL), so I believe that information needs to be passed to the front end regardless? I do agree though that the selectedRepoName bit could be cleaner, I'll fix that now.